Backtracking/MinMax.js

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *  Min Max problem
+ *  See https://www.geeksforgeeks.org/minimax-algorithm-in-game-theory-set-1-introduction/
+ *  Min Max problem is a problem where we need to find the maximum score of an event by comparing all * *  possible moves in the game
+ *
+ */
+
+//     nodeIndex is index of current node in scores[].
+//     if move is of maximizer return true else false
+//     leaves of game tree is stored in scores[]
+//     this.height is maximum this.height of Game tree
+
+//     depth is current depth in game tree.
+//     nodeIndex is index of current node in scores[].
+//     scores[] contains the leaves of game tree.
+//     this.height is maximum this.height of game tree
+
+//     >>> scores = [90, 23, 6, 33, 21, 65, 123, 34423]
+//     >>> this.height = math.log(len(scores), 2)
+//     >>> minimax(0, 0, True, scores, this.height)
+//     65
+//     >>> minimax(-1, 0, True, scores, this.height)
+//     Traceback (most recent call last):
+//         ...
+//     ValueError: Depth cannot be less than 0
+//     >>> minimax(0, 0, True, [], 2)
+//     Traceback (most recent call last):
+//         ...
+//     ValueError: Scores cannot be empty
+//     >>> scores = [3, 5, 2, 9, 12, 5, 23, 23]
+//     >>> this.height = math.log(len(scores), 2)
+//     >>> minimax(0, 0, True, scores, this.height)
+//     12
+
+class MinMax {
+  constructor (scores, depth, nodeIndex, isMax) {
```

Comment:
This constructor is completely superfluous - you're passing all these as arguments to `solve` later on. I'm questioning whether this should be a class at all - IMO it should just be a single `solve` function that checks it parameters and recurses.